pmd-core/src/main/java/net/sourceforge/pmd/RuleSetWriter.java

```diff
@@ -266,21 +265,11 @@ private Element createPropertiesElement(List<PropertyDescriptor<?>> propertyDesc
         Element propertiesElement = null;
         if (propertyDescriptors != null) {
 
-            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) { // For
-                // each
-                // provided
-                // PropertyDescriptor
-
-                if (propertyDescriptor instanceof PropertyDescriptorWrapper) { // Any
-                    // wrapper
-                    // property
-                    // needs
-                    // to
-                    // go
-                    // out
-                    // as
-                    // a
-                    // definition.
+            for (PropertyDescriptor<?> propertyDescriptor : propertyDescriptors) {
+                // For each provided PropertyDescriptor
+
+                if (propertyDescriptor instanceof PropertyDescriptorWrapper) {
```

Comment:
This would seem to be the root of a _lot_ of complexity.  I can no long remember the full subtleties which led me to introduce the wrapper, and my comments here are insufficient to jog my memory.  I checked my email history too.  Let's assume the comment on `PropertyDescriptorWrapper` tells the whole story (not sure what's going on in the UIs, @hooperbloob  always knew far more about that than me), and this is truly just about knowing whether to serialize as a _value_ or a _descriptor_.

Looking at the code I have local, without these changes, I'd think if the _creation_ of a `PropertyDescriptor` by the `RuleSetFactory`, _or_ the subsequent `Rule.definePropertyDescriptor` could cause some _stateful_ information to be set onto the descriptor or the Rule, it could be used to obviate the need for wrappers.  The former seems more preferable.  So...

What if `PropertyDescriptor` had a _source_ specified upon creation?  Say, INTERNAL or EXTERNAL?  Java based Rules would use INTERNAL, and XML would use EXTERNAL.  And thus, INTERNAL means serialize as value, and EXTERNAL means serialize as descriptor.

Then everything wrapper related goes the way of the telegraph.